deploying/metadatastore.rst

```diff
@@ -0,0 +1,47 @@
+Deploying metadatastore
+-----------------------
+
+Lessons learned from SRX and CSX
+================================
+
+Beamlines will have /DATA and /POOL data storage. /DATA seems to be for
+experimental data and /POOL for analysis.  SRX and CSX have opted to run the
+mongo database daemon on the channel archiver server (xf05id-ca1, xf23id-ca).
+This seems to be a reasonable setup and will be repeated on other beamlines
+until someone provides a better option.  We found it useful to set up an alias
+as CSX called 'xf23id-broker' that metadatastore and filestore could point to
+so that they dont actually care which machine the mongodb daemon is running on.
+We should do the same for SRX and all future beamlines, unless a better option
+is presented.  Contact Petkus to do this.
+
+Proposed Deployment Strategy
+============================
+
+#. Set up an alias at beamline for the server running the mongo daemon
+   (xf##id-broker)
+#. Install and configuration of mongodb on xf##id-broker
+
+    #. Install mongodb on xf##id-broker::
+
+        sudo apt-get install mongodb
+
+    #. Edit /etc/mongodb.conf so that ``dbpath=/DATA/mongodb/`` and
+       ``bind_ip = 0.0.0.0``
```

Comment:
Agreed,

Petkus should do this with puppet so that the /etc/mongodf.conf is managed centrally. He should also watch diskspace on the /DATA to ensure we are not getting into problems.

S
On Feb 12, 2015, at 11:04 AM, Thomas A Caswell <notifications@github.com<mailto:notifications@github.com>> wrote:


In deploying/metadatastore.rst<https://github.com/NSLS-II/docs/pull/5#discussion_r24591836>:

> +We should do the same for SRX and all future beamlines, unless a better option
> +is presented.  Contact Petkus to do this.
> +
> +Proposed Deployment Strategy
> +============================
> +
> +#. Set up an alias at beamline for the server running the mongo daemon
> +   (xf##id-broker)
> +#. Install and configuration of mongodb on xf##id-broker
> +
> +    #. Install mongodb on xf##id-broker::
> +
> +        sudo apt-get install mongodb
> +
> +    #. Edit /etc/mongodb.conf so that ``dbpath=/DATA/mongodb/`` and
> +       ``bind_ip = 0.0.0.0``


Maybe this should be a bit more locked I think.

—
Reply to this email directly or view it on GitHub<https://github.com/NSLS-II/docs/pull/5/files#r24591836>.